lib/api/kuzzle.js

```diff
@@ -284,43 +270,37 @@ function registerErrorHandlers(kuzzle) {
       kuzzle.log.error(`ERROR: unhandledRejection: ${promise}`);
     }
 
-    // dump+exit is a good idea during development as it helps
-    // spotting code errors. But in production, these problems
-    // should only come from plugins, and it seems a really bad idea
-    // to stop Kuzzle because a plugin created an unchained, uncatchable
-    // promise that got rejected
-    if (process.env.NODE_ENV !== 'production') {
-      request.input.args.suffix = 'unhandled-rejection';
-      kuzzle.adminController.dump(request)
-        .finally(() => process.exit(1));
+    // dump+exit on an unhandled rejection is a good idea during development
+    // as it helps // spotting code errors (even if automated dumps
+    // are disabled).
+    // But in production, these problems // should only come from plugins,
+    // and it seems a really bad idea to stop Kuzzle because a plugin created
+    // an unchained, uncatchable promise that got rejected
+    if (process.env.NODE_ENV === 'development') {
+      kuzzle.janitor.dump('unhandled-rejection').finally(() => process.exit(1));
```

Comment:
I disagree to force dumps in development environments. More over, I generally consider the dump feature to be not only useless but harmful in production and an annoyance during development.

I understand we cannot remove the feature on the current version but I would definitely not force them in any situation.

Reply:
It depends what you means by "dump": if you mean the automated core dump, then this is debatable. This didn't not occur yet, but if we do have random problems, this can be handy. And automated dumps should only occur when kuzzle crash (if they occur elsewhere, then this needs to be fixed) and in those cases, by defintion, we might not have enough logs allowing support actions on our sides.
Anyway, we can debate the usefulness of core dumps.

But collecting basic informations about the system (node & npm versions, kuzzle + es + redis versions, logs) in an easy way for clients to send to us for support actions cannot be useless.

Reply:
One more thing: you miss the point of this PR. This does not reactivate automated dumps, they still work if you activate the option. What does not work anymore is everything else when automated dumps are disabled: graceful shutdown, manual dumps and other global exceptions/rejections handlers.

Reply:
About forced dumps during development, I agree that letting Kuzzle crash should be enough, especially now that we have chrome dev tools to intercept uncaught exceptions.

I'll remove that part.